src/main/java/core/basesyntax/model/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.model.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.OperationHandler;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private static final int DEFAULT_AMOUNT = 0;
+    private final Storage storage;
+
+    public PurchaseOperationHandler(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        int currentAmount = storage.getInventory().getOrDefault(transaction.getFruit(),
+                DEFAULT_AMOUNT);
+        storage.getInventory().put(transaction.getFruit(),
+                currentAmount - transaction.getQuantity());
```

Comment:
is it correct logic? can you purchase fruits if it doesn't exist in storage or the amount isn't enough?